packages/docs/pages/Tree/TreePage.tsx

```diff
@@ -0,0 +1,143 @@
+              },
+            ],
+          },
+          {
+            id: 1,
+            value: 1,
+            label: 'Category',
+            disabled: true,
+            expanded: true,
+            children: [{ id: 6, value: 6, label: 'Category' }],
+          },
+          { id: 2, value: 2, label: 'Category' },
+          { id: 3, value: 3, label: 'Category', selected: true, children: [{ id: 7, value: 7, label: 'Category' }] },
+          { id: 4, value: 4, label: 'Category', children: [{ id: 8, value: 8, label: 'Category' }] },
+        ];
+
+        return <Tree selectable="multi" nodes={nodes} />;
+      }}
+      {/* jsx-to-string:end */}
+    </CodePreview>
+
+    <H1>API</H1>
+    <TreePropTable />
+    <TreeNodePropTable id="tree-node-prop-table" />
+
+    <H1>Examples</H1>
+
+    <H2>Radio</H2>
+    <CodePreview>
+      {/* jsx-to-string:start */}
+      {function Example() {
+        const nodes = [
+          {
+            id: 0,
+            value: 0,
+            label: 'Category',
+            expanded: true,
+            disabled: true,
+            children: [{ id: 3, value: 3, label: 'Subcategory' }],
+          },
+          { id: 1, value: 1, label: 'Category' },
+          { id: 2, value: 2, label: 'Category', disabled: true },
+        ];
+
+        return <Tree selectable="radio" nodes={nodes} />;
+      }}
+      {/* jsx-to-string:end */}
+    </CodePreview>
+
+    <H2>Custom Icons</H2>
+    <Text>You can replace the folder icon with a custom icon of your choice.</Text>
+    <Message
+      type="warning"
+      messages={[{ text: 'If you use the iconless prop, ALL icons with be hidden (including custom ones).' }]}
+      marginBottom="medium"
+    />
+    <CodePreview>
+      {/* jsx-to-string:start */}
+      {function Example() {
+        const nodes = [
+          {
+            id: 0,
+            icon: <StoreIcon color="primary" />,
+            label: 'Storefront - US',
+            expanded: true,
+            children: [{ id: 3, label: 'Subcategory' }],
+          },
+          { id: 1, icon: <LanguageIcon color="primary" />, label: 'Storefront - CA' },
+          { id: 2, icon: <AssignmentIcon color="primary" />, label: 'Storefront - EU' },
+        ];
+
+        return <Tree nodes={nodes} />;
+      }}
+      {/* jsx-to-string:end */}
+    </CodePreview>
+
+    <H2>Modifying Children</H2>
+    <Text>
+      By returning a object containing new children nodes, you have the freedom of modifying/replacing children nodes.
+      This is useful when you need to asyncronously load in child nodes.
+    </Text>
+    <CodePreview>
+      {/* jsx-to-string:start */}
+      {function Example() {
+        const nodes = [
+          {
+            id: 0,
+            label: 'Category',
+            children: [],
+          },
+          { id: 1, label: 'Category' },
+          { id: 2, label: 'Category' },
+        ];
+
+        const handleOnExpand = ({ children }: TreeNodeProps<undefined>) => {
```

Comment:
What do you think about changing this example a bit to showcase a mock async API request with a node id? You could define the "logic" outside of the example so that it doesn't render in the docs and then just use it like... `await fetchChildren(node.id)`.

It should help showcase that you get the full `node` as params and that you can return a promise.